src/consensus/consensus.rs

```diff
@@ -73,7 +73,18 @@ impl Config {
     }
 
     pub fn num_shards(&self) -> u32 {
-        self.shard_ids.len() as u32
+        self.shard_ids().len() as u32
```

Comment:
This was why it wasn't working for multiple shards it was using "1,2".len() instead of vec![1,2].len(). So the Block proposer was awaiting for a 3rd chunk that would never arrive.